FEATURE: New content

Sebobo opened the pull request:
Depends on #137

jonnitto committed:
TASK: Switch to postcss and esbuild

jonnitto committed:
TASK: Switch to markdown for readme

jonnitto committed:
Update: Remove readme

jonnitto committed:
Update: Re-add readme (Uppercase)

jonnitto committed:
Update: Reduce types of input files

jonnitto committed:
TASK: Suggest monocle and Yoast SEO

jonnitto committed:
TASK: Add tailwindCSS and AlpineJS

jonnitto committed:
TASK: Edit comment on not found page

jonnitto committed:
TASK: Change default values and icon from not found document

jonnitto committed:
TASK: Update Youtube element

jonnitto committed:
TASK: Add Javascript and Stylesheet atoms

jonnitto committed:
TASK: Add JS and CSS to Monocle Preview page

jonnitto committed:
TASK: Add fonts and basic css

jonnitto committed:
TASK: Remove old youtube integration

jonnitto committed:
TASK: Add AlpineJS

jonnitto committed:
TASK: Add Breadcrumb

jonnitto committed:
!!! TASK: Add headline

Breaking because the tag name is now set in the inspector

jonnitto committed:
TASK: Add node presets

jonnitto committed:
!!! TASK: Add headline

Breaking because the tag name is now set in the inspector

jonnitto committed:
TASK: Add image

jonnitto committed:
BUGFIX: Add missing monocle comment

jonnitto committed:
FIX: Remove double setting fontSize

jonnitto committed:
TASK: Reduce font-weight for bold

jonnitto committed:
TASK: Add text element

jonnitto committed:
TASK: Add monocle settings

jonnitto committed:
FIX: Add missing property

jonnitto committed:
TASK: Fix indentation

jonnitto committed:
TASK: Add text with image

jonnitto committed:
TASK: Remove old files

jonnitto committed:
TASK: Add alert atom

jonnitto committed:
TASK: Add columns element

jonnitto committed:
TASK: Remove print preview and add responsive preview

jonnitto committed:
TASK: Move document fusion files to nodetypes folder

jonnitto committed:
Merge branch 'master' of github.com:neos/Neos.Demo into update/frontend_build_stack

jonnitto committed:
TASK: Remove moved files

jonnitto committed:
FEATURE: Add code editor to ck inline

jonnitto committed:
TASK: Switch to variable font

jonnitto committed:
TASK: Add header presentation

jonnitto committed:
TASK: Add comment about monocle

jonnitto committed:
TASK: Use Grid in header also in desktop view

Sebobo committed:
TASK: Split settings files

Sebobo committed:
TASK: Update site export

Sebobo committed:
TASK: Cleanup constraints

They are now everywhere defined via constraint mixins

Sebobo committed:
FEATURE: Define meta menu items via homepage

Sebobo committed:
BUGFIX: Make the meta nav work again and cache globally

Sebobo committed:
TASK: Update main banner text

jonnitto committed:
BUGFIX: Renderer in two columns

jonnitto committed:
TASK: Add base template

jonnitto committed:
TASK: Improve typography

jonnitto committed:
TASK:: Improve header rendering

jonnitto committed:
TASK: Add basic document integration

jonnitto committed:
TASK: Enable alpine devtools in dev mode

jonnitto committed:
TASK: Intagrate header

jonnitto committed:
TASK: Intagrate landing page

jonnitto committed:
TASK: Remove old js and css files

jonnitto committed:
TASK: Remove icon- from icon names

jonnitto committed:
TASK: Optimze document nodes

* Allow notfound on homepage
* Hide notfound from creation dialog
* Add document mixin
* Remove teaser image from default pages

jonnitto committed:
TASK:: Remove only onced used prototype

jonnitto committed:
TASK: Add Footer

jonnitto committed:
TASK: Move header integration file

jonnitto committed:
TASK: Enable alternate language links

* We use this for our language menu

jonnitto committed:
BUGFIX: Render columns only if there is content

jonnitto committed:
TASK: Integrate breadcrumb

jonnitto committed:
BUGFIX: Check for special case for state for home link

jonnitto committed:
TASK: Optimize CSS for layout

jonnitto committed:
TASK: Remove defaultUriSuffix

jonnitto committed:
TASK: Add chapter menu

jonnitto committed:
TASK: Use content for prop in text atom

jonnitto committed:
TASK: Check if the menu has items

jonnitto committed:
TASK: Set correct position for content collection

It reflects now the UI

jonnitto committed:
TASK: Move main collection fusion

jonnitto committed:
TASK: Remove old files

jonnitto committed:
TASK: Sort content elements in creation dialog

jonnitto committed:
TASK: Rename column presentation element

jonnitto committed:
TASK: Move contact form and regisration form

jonnitto committed:
TASK: Remove old files

jonnitto committed:
TASK: Removee chapter menu content and add chapter container document

jonnitto committed:
TASK: Set correct name for main content collection

jonnitto committed:
TASK: OPtimize base mixins

jonnitto committed:
TASK: Add entry point for custom colors

jonnitto committed:
TASK: Add chapter

jonnitto committed:
TASK: Add constraint mixins

jonnitto committed:
TASK: Add new carousel

jonnitto committed:
TASK: Make usage of showInCreationDialog

jonnitto committed:
TASK: Rename Chapter Container to Book

jonnitto committed:
TASK: Make usage of document constaints

jonnitto committed:
TASK: Remove not used plugins

jonnitto committed:
TASK: Better entrypoint for own logo

jonnitto committed:
TASK: Render CSS and JS

Sebobo committed:
Merge branch 'update/frontend_build_stack' into feature/new-content

# Conflicts:
#	Configuration/Settings.yaml
#	NodeTypes/Content/Carousel.yaml
#	NodeTypes/Content/ChapterMenu.yaml
#	NodeTypes/Content/ContactForm/ContactForm.yaml
#	NodeTypes/Content/Headline.yaml
#	NodeTypes/Content/Image/Image.yaml
#	NodeTypes/Content/Registration/Registration.yaml
#	NodeTypes/Content/Text/Text.yaml
#	NodeTypes/Content/TextWithImage/TextWithImage.yaml
#	NodeTypes/Content/YouTube/YouTube.yaml
#	NodeTypes/Document/Chapter/Chapter.yaml
#	NodeTypes/Document/Homepage/Homepage.yaml
#	NodeTypes/Document/LandingPage.yaml
#	NodeTypes/Document/Page.yaml
#	NodeTypes/Override/AssetList.yaml
#	NodeTypes/Override/ContentReferences.yaml
#	Resources/Private/Fusion/Content/ChapterMenu.fusion
#	Resources/Private/Fusion/Document/Chapter.fusion
#	Resources/Private/Fusion/Document/Fragment/Menu/Meta.fusion
#	Resources/Private/Fusion/Main.js
#	Resources/Private/JavaScript/main.js
#	Resources/Public/Scripts/Main.js
#	Resources/Public/Scripts/Main.js.map

Sebobo committed:
TASK: Remove fluid templates

Sebobo committed:
TASK: Use dark color for the hero background

Sebobo committed:
TASK: Refactor footer and header prototype

Sebobo committed:
TASK: Adjust content to new nodetypes

jonnitto committed:
TASK: Update content

* Fix headlines
* Add youtube slide to carousel
* Fix breaks in the book chalters

jonnitto committed:
BUGFIX: Allow Youtube element

jonnitto committed:
BUGFIX: Set mac width for image in text with image

jonnitto committed:
BUGFIX: some margins

jonnitto committed:
BUGFIX: Adjust margin for paragraphs and figures

jonnitto committed:
TASK: Add a few print styles

#144

regniets committed:
BUGFIX: Added missing labels

regniets reviewed (approved):
Checked through your changes - Kudos, really like them.
Concerning demo.neos.io, i am ok with merging. 
I really like the code style editor plugin!
Not quite sure about the rest - what else is the package being used for?

jonnitto committed:
BUGFIX: Margin between element

Grid don't collapse margins, so it needs a solution without it

jonnitto committed:
BUGFIX: Wait for check if hamburger button is hidden

jonnitto committed:
TASK: Improve form rendering

 #144

jonnitto committed:
BUGFIX: Remove print editpreview config

We have no print view anymore

jonnitto committed:
BUGFIX: Some edge cases with the main menu

jonnitto committed:
TASK: Render CSS and JS

jonnitto committed:
TASK: Move headline outside of column

jonnitto committed:
TASK: Update AlpineJS

jonnitto committed:
TASK: Update frontend packages

jonnitto committed:
TASK: Use localized label for german language

jonnitto committed:
TASK: Use CR settings for language menu

jonnitto committed:
TASK: Fix some styling issues with form and set default classes

jonnitto committed:
TASK: Add attributes to form field

jonnitto committed:
TASK: Add id and action to forms

Sebobo committed:
TASK: Update code to use PHP8 features

Sebobo committed:
FEATURE: Remove name requirement for registration form

Sebobo committed:
FEATURE: Validate for existing usernames during registration

jonnitto committed:
Update: Rename presenation components

jonnitto committed:
TASK: Render JS and CSS, upgrade packages

jonnitto committed:
TASK: Document code and add some todo comments

jonnitto committed:
TASK: Rename ckeditor plugin

jonnitto committed:
TASK: Remove document mixin

jonnitto committed:
TASK: Add labels for screenreader

ahaeslich commented:
@jonnitto, @Sebobo  Do we have an overview what still needs to be done here? And are we aiming to include the refactoring for the 8.3 release?

Sebobo commented:
I really want this for 8.3 as it would also be nice for the NeosCon and the contribution day.

jonnitto committed:
Upgrade: Neos 8.2

jonnitto committed:
Update: Use uri from Menuitems

Sebobo committed:
FEATURE: Use allow list constraints instead of restrictions

ahaeslich commented on Resources/Private/Fusion/Root.fusion:
```diff
@@ -1,4 +1,7 @@
+include: **/*.fusion
+// TODO Remove this line when the following PR is merged:
+// https://github.com/neos/neos-development-collection/pull/3903
+include: ../../../NodeTypes/**/*.fusion
```
The pull request was merged. We should change this line.
It should work with `include: nodetypes://Neos.Demo/**/*.fusion`

ahaeslich committed:
Merge branch '8.3' into feature/new-content

# Conflicts:
#	Resources/Private/Content/Sites.xml
#	composer.json

mficzel committed:
TASK: Restructure nodetypes

- Rename `Neos.Demo:Document.Mixin.AbstractDocument` to `Neos.Demo:Document`
- Move text presets into `neosdemo` namespace
- Adjust constraint to only allow `Neos.Demo:Document.SubPage` generally
- Adjust fusion include from NodeTypes folder to new `nodetypes://`
- Move `Neos.Demo:Layout.Page` and `Neos.Demo:Layout.Page` into `Presentation` namespace
- Move classes `content` and `prose` to `Presentation.Layout.*` prototypes

mficzel committed:
TASK: Restructure document rendering

- Rename Neos.Demo:DocumentRenderer to Neos.Demo:Document
- Remove Augmenter override and Page override
- Introduce separate landing page layout prototype

mficzel committed:
TASK: Adjust the registration content to inherit from Neos.Neos:ContentComponent for better editor interaction

mficzel committed:
TASK: Adjustments and temporary disabeling of ck-plugin

mficzel committed:
FEATURE: DocumentTypes Blog and BlogPosting with basic rendering

mficzel committed:
TASK: Add authorName and make this and datePublished required + showInCreationDialog

mficzel committed:
TASK: Move blog and group documents to new nodetype group special

mficzel committed:
TASK: Simplify rendering of chapter and book documents

Sebobo committed:
BUGFIX: Make codeplugin work again

Sebobo committed:
TASK: Show landingpage in special group

Sebobo committed:
!!! TASK: Remove book and chapter code

Is replaced by the new blog and article types

Sebobo commented:
@ahaeslich @mficzel are we done here now? 

We might need to update the content export with the blog pages so we don't have inconsistencies, but is there anything else?

mficzel commented:
Content for the blog would be great and I think we wanted to add some BlogList content element

mficzel committed:
FEATURE: Add content `Content.BlogPostingList` and introduce `Cards.Container` and `Cards.Cart` on the presentation side

mficzel commented:
I pushed a new commit wich adds `Content.BlogPostingList`and introduces `Cards.Container` and `Cards.Cart` on the presentation side`

Both are still ugly even though i tried to use the card components from https://tailwind-elements.com/docs/standard/components/cards/ .. so someone with little tailwind knowhow and a non-backend eye should adjust those.

mficzel committed:
TASK: Add labels to `Content.BlogPostingList`

ahaeslich committed:
TASK: Update styling of blog listing

- Use title prop as headline
- Fix card / card container styling

mficzel committed:
TASK: Add cache annotations and adjust order of blog postings

ahaeslich committed:
TASK: Update styling of blog posting

ahaeslich committed:
TASK: Update demo content with blog pages

ahaeslich commented:
> @ahaeslich @mficzel are we done here now?
> 
> We might need to update the content export with the blog pages so we don't have inconsistencies, but is there anything else?

Done ✅

ahaeslich commented:
@Sebobo & @mficzel I think we can merge this now and add all improvements in smaller merge requests

I would check and add #158 afterwards.

mficzel reviewed (approved):
Agree to get this in now and fix remaining issues later. Actually like the state but would like to know what @Sebobo and @jonnitto say.

mhsdesign commented:
Happy to see that we are on the final track ;)

i guess Jon and Seb will have a closer look at the demo on the neos to 100 day and if they find something we can still correct it then ;)

mficzel committed:
TASK: Tune cache tags for blog posting lists

mficzel commented:
Yes I would not wait for feedback but give them some days.

mficzel commented:
Hmm ... this change breaks functional tests in the neos-development collection because of dropping the .html suffix. Do we want this that much to adjust the tests or do we want to revert that part.

```
2) Neos\Neos\Tests\Functional\Service\LinkingServiceTest::linkingServiceCreatesUriViaGivenNodeObject
Failed asserting that '/en/home' ends with "en/home.html".

/var/www/html/Packages/Neos/Neos.Neos/Tests/Functional/Service/LinkingServiceTest.php:336
/var/www/html/Packages/Neos/Neos.Neos/Tests/Functional/Service/LinkingServiceTest.php:149
/var/www/html/bin/phpunit:123

3) Neos\Neos\Tests\Functional\Service\LinkingServiceTest::linkingServiceCreatesUriViaAbsoluteNodePathString
Failed asserting that '/en/home' ends with "en/home.html".

/var/www/html/Packages/Neos/Neos.Neos/Tests/Functional/Service/LinkingServiceTest.php:336
/var/www/html/Packages/Neos/Neos.Neos/Tests/Functional/Service/LinkingServiceTest.php:157
/var/www/html/bin/phpunit:123

4) Neos\Neos\Tests\Functional\Service\LinkingServiceTest::linkingServiceCreatesUriViaStringStartingWithTilde
Failed asserting that '/en/home' ends with "en/home.html".

.... and so on

```

mhsdesign commented:
@mficzel  i think we should fix the tests instead of reintroducing the suffix ^^ (discussed with @ahaeslich)
